src/tileset.rs

```diff
@@ -67,6 +94,18 @@ impl Tileset {
             },
         });
 
+        let columns = match columns {
+            Some(col) => col,
+            None => match &image {
+                None => {
+                    return Err(TiledError::MalformedAttributes(
+                        "No <image> and no <columns> in <tileset>".to_string(),
```

Comment:
Message is a bit confusing because `columns` is an attribute rather than an element.